src/wp-includes/formatting.php

```diff
@@ -3358,6 +3358,13 @@ function convert_smilies( $text ) {
 	if ( get_option( 'use_smilies' ) && ! empty( $wp_smiliessearch ) ) {
 		// HTML loop taken from texturize function, could possible be consolidated.
 		$textarr = preg_split( '/(<.*>)/U', $text, -1, PREG_SPLIT_DELIM_CAPTURE ); // Capture the tags as well as in between.
+
+		if ( preg_last_error() != PREG_NO_ERROR ) {
```

Comment:
```suggestion
		if ( preg_last_error() !== PREG_NO_ERROR ) {
```

Should we use strict checking here? Note that  `PREG_NO_ERROR` is `0`.